Snowflake.Data/Core/FileTransfer/StorageClient/SFS3Client.cs

```diff
@@ -266,26 +267,38 @@ private GetObjectRequest GetFileHeaderRequest(ref AmazonS3Client client, SFFileM
         /// <param name="fileMetadata">The S3 file metadata.</param>
         /// <param name="response">The Amazon S3 response.</param>
         /// <returns>The file header of the S3 file.</returns>
-        private FileHeader HandleFileHeaderResponse(ref SFFileMetadata fileMetadata, GetObjectResponse response)
+        internal FileHeader HandleFileHeaderResponse(ref SFFileMetadata fileMetadata, GetObjectResponse response)
         {
             // Update the result status of the file metadata
             fileMetadata.resultStatus = ResultStatus.UPLOADED.ToString();
 
             SFEncryptionMetadata encryptionMetadata = new SFEncryptionMetadata
             {
-                iv = response.Metadata[AMZ_IV],
-                key = response.Metadata[AMZ_KEY],
-                matDesc = response.Metadata[AMZ_MATDESC]
+                iv = GetMetadataCaseInsensitive(response.Metadata, AMZ_IV),
+                key = GetMetadataCaseInsensitive(response.Metadata, AMZ_KEY),
+                matDesc = GetMetadataCaseInsensitive(response.Metadata, AMZ_MATDESC)
             };
 
             return new FileHeader
             {
-                digest = response.Metadata[SFC_DIGEST],
+                digest = GetMetadataCaseInsensitive(response.Metadata, SFC_DIGEST),
                 contentLength = response.ContentLength,
                 encryptionMetadata = encryptionMetadata
             };
         }
 
+        private string GetMetadataCaseInsensitive(MetadataCollection metadataCollection, string metadataKey)
+        {
+            var value = metadataCollection[metadataKey];
+            if (value != null)
+                return value;
+            if (string.IsNullOrEmpty(metadataKey))
+                return null;
+            var keysCaseInsensitive = metadataCollection.Keys
+                .Where(key => $"x-amz-meta-{metadataKey}".Equals(key, StringComparison.OrdinalIgnoreCase));
```

Comment:
Therre was no prefixing with `x-amz-meta-` before, why is it now?

Reply:
The class MetadataCollection automatically adds "x-amz-meta-" prefix to all its elements.
So if you would like to store `X-AMZ-IV` in this collection, you are really storing for the key of 'x-amz-meta-X-AMZ-IV'. That's why when I would like to find a key which is matching to `X-AMZ-IV` I need to look for 'x-amz-meta-X-AMZ-IV'.
MetadataCollection.MetaDataHeaderPrefix prefix is internal so I couldn't take the value of prefix from the AWS library and I had to duplicate the prefix value here.